.github/workflows/release-build.yml

```diff
@@ -19,25 +23,42 @@ jobs:
   create-all-submodule-archive:
     name: Create all submodule archive
     runs-on: ubuntu-latest
-    if: (github.event.action == 'published') || (github.event.action == 'helics-release-build' && (contains(github.event.client_payload.packages, 'archive') || contains(github.event.client_payload.packages, 'everything'))) || github.event_name == 'workflow_dispatch'
+    if: (github.event.action == 'published') || (github.event_name == 'workflow_dispatch' && (contains(github.event.inputs.packages, 'archive') || contains(github.event.inputs.packages, 'everything')))
     steps:
-    - name: Checkout event ref
-      uses: actions/checkout@v2
+    - name: Determine fetch depth
+      id: determine_fd
+      shell: bash
+      run: |
+        unset DEPTH
+        if [ -n '${{ github.event.inputs.commit }}' ]; then DEPTH='0'; else DEPTH='1'; fi
+        echo "::set-output name=DEPTH::${DEPTH}"
+
+    - name: Checkout code
+      uses: actions/checkout@v3
```

Comment:
are there other places we need to move to v3 of the checkout action?

Reply:
It just uses a newer version of node.js, no major changes and v2 still works fine. I'm looking at setting up Dependabot to keep the versions of actions used updated so that should take care of upgrading other places eventually.